component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2114,6 +2115,40 @@ protected void triggerEvent(String userName, String tenantDomain, String userSto
         }
     }
 
+    /**
+     * Method to Trigger the Email otp event.
+     *
+     * @param userName            Identity user name.
+     * @param tenantDomain        Tenant domain of the user.
+     * @param userStoreDomainName User store domain name of the user.
+     * @param notificationEvent   The name of the event.
+     * @param otpCode             The OTP code returned for the authentication request.
+     * @param sendToAddress       The email address to send the otp.
+     * @param serviceProvider     Name of the service provider that the user is trying to login.
+     * @throws AuthenticationFailedException In occasions of failing to send the email to the user.
+     */
+    protected void triggerEvent(String userName, String tenantDomain, String userStoreDomainName,
+                                String notificationEvent, String otpCode, String sendToAddress, String serviceProvider)
+            throws AuthenticationFailedException {
+
+        String eventName = IdentityEventConstants.Event.TRIGGER_NOTIFICATION;
+        HashMap<String, Object> properties = new HashMap<>();
```

Comment:
why we use object type

Reply:
@Buddhimah that is the way that we pass event properties. this enables us to pass any object.

Reply:
its a local variable of a protected method( cant change signature later) only parameter type I can see is Strings?

Reply:
I don't think we need to make this `Protected`.  I think she has used this to maintain consistency.

Reply:
When initializing an event, it expects a `Map<String, Object> eventProperties`. Hence object type has to be used here.